core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1073,6 +1073,9 @@ else if (ALT_ROUTE.equalsIgnoreCase(algoStr))
                 return Collections.emptyList();
 
             QueryGraph queryGraph = QueryGraph.lookup(graph, qResults);
+            if (weighting instanceof BlockAreaWeighting)
+                ((BlockAreaWeighting) weighting).setQueryGraph(queryGraph);
```

Comment:
Here we could think about a more generic solution as this is also required for the CustomWeighting in #1841.